src/Image.cc

```diff
@@ -766,6 +766,48 @@ static void jpeg_mem_src (j_decompress_ptr cinfo, void* buffer, long nbytes) {
 
 #endif
 
+class BufferReader : public Image::Reader {
+public:
+  BufferReader(uint8_t* buf, unsigned len) : _buf(buf), _len(len), _idx(0) {}
+
+  bool hasBytes(unsigned n) const override { return (_idx + n - 1 < _len); }
+
+  uint8_t getNext() override { return _buf[_idx++]; }
```

Comment:
Should this be doing a bounds check? I see that you're always calling `hasBytes` before `getNext`, but wonder if it's better for future changes to guard this. `getc` below essentially does bounds checking, I think.

Reply:
Personally, I think there's no point doing unnecessary bounds checking, but I've added it.